profiling/.gitignore

```diff
@@ -0,0 +1,9 @@
+flamegraph*
+*folded
+*txt
+*dat
+*json
+*png
+memory-profiler-*
+libmemory_profiler.so
+perf.data
```

Comment:
This file makes it so that these files are difficult to delete en masse (i.e. `git clean -df` doesn't get them). Can we add a `clean-profile` target to the Makefile to ensure these files are removed?

Reply:
done in 99b8d8cb8461fdeb9e25bfeab63813c38ff49efc